From 0f7af311bf5e7e97fde68fc24fc852c7f50ee882 Mon Sep 17 00:00:00 2001
From: Teemu Ollakka <teemu.ollakka@galeracluster.com>
Date: Tue, 1 Nov 2022 17:09:39 +0200
Subject: [PATCH] Fix memory leak in gcs gcomm backend

The tp_ object is not deleted in GCommConn destructor, which leaks
memory if gcomm backed open fails (e.g. address already in use),
which in turn causes excessive leak reports when running with ASAN build.

In order to suppress excessive memory leak reports, delete tp_
object in GCommConn destructor.

codership/galera-bugs#834
---
 gcs/src/gcs_gcomm.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcs/src/gcs_gcomm.cpp b/gcs/src/gcs_gcomm.cpp
index 2953c9afb..80a0431e8 100644
--- a/gcs/src/gcs_gcomm.cpp
+++ b/gcs/src/gcs_gcomm.cpp
@@ -164,6 +164,7 @@ class GCommConn : public Toplay
 
     ~GCommConn()
     {
+        delete tp_;
         delete net_;
     }
 
